api/app/signals/apps/history/middleware.py

```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2022 Gemeente Amsterdam
+from signals.apps.history.services import HistoryLogService
+
+
+class HistoryMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        HistoryLogService.thread.request = request
```

Comment:
I agree and see the issue.

First I did not fully realized that I not always have the request. For example a Django signal cannot access the request. Therefore I made the decision to implement the same behaviour as in the `change_log` app because this was a lot of scope creep. I created a ticket on our backlog to refactor this to a better solution.

For now I added additional checks in the middleware that the request is only added when making a POST/PUT/PATCH request on the endpoints for changing a category and user AND the admin pages. This would limit the change that a race condition can occur.